src/w3iProxy/pushProviders/externalPushProvider.ts

```diff
@@ -2,18 +2,34 @@ import type { EventEmitter } from 'events'
 import type { JsonRpcRequest, JsonRpcResult } from '@walletconnect/jsonrpc-utils'
 import { formatJsonRpcRequest } from '@walletconnect/jsonrpc-utils'
 import type { PushClientFunctions, W3iPushProvider } from './types'
+import { ExternalCommunicator } from '../externalCommunicators/communicatorType'
+import { AndroidCommunicator } from '../externalCommunicators/androidCommunicator'
+import { IOSCommunicator } from '../externalCommunicators/iosCommunicator'
+import { JsCommunicator } from '../externalCommunicators/jsCommunicator'
 
 export default class ExternalPushProvider implements W3iPushProvider {
   protected readonly emitter: EventEmitter
   private readonly methodsListenedTo = ['push_request', 'push_message']
   public providerName = 'ExternalPushProvider'
+  protected readonly communicator: ExternalCommunicator
 
   /*
    * We have no need to register events here like we do in internal provider
    * because the events come through the emitter anyway.
    */
-  public constructor(emitter: EventEmitter) {
+  public constructor(emitter: EventEmitter, name: string) {
     this.emitter = emitter
+    switch (name) {
+      case 'android':
+        this.communicator = new AndroidCommunicator(this.emitter)
+        break
+      case 'ios':
+        this.communicator = new IOSCommunicator(this.emitter)
+        break
+      default:
+        this.communicator = new JsCommunicator(this.emitter)
+        break
+    }
   }
 
   protected async postToExternalProvider<MName extends keyof PushClientFunctions>(
```

Comment:
Should this also delegate to the `communicator` here as for `externalChatProvider`? https://github.com/WalletConnect/web3inbox/pull/51/files#diff-1f7b1977dc91a5064fee19ccb242c192b43705ceb88616ca0c87a7ad61a8091eR48